pkg/apis/kubeone/validation/validation.go

```diff
@@ -731,59 +728,45 @@ func ValidateHelmReleases(helmReleases []kubeoneapi.HelmRelease, fldPath *field.
 		}
 
-		if (clusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv6 || clusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv4IPv6 || clusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv6IPv4) && len(h.IPv6Addresses) == 0 {
+		if (clusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv6 || clusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv4IPv6 || clusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv6IPv4) && len(host.IPv6Addresses) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath, "no IPv6 address given"))
 		}
-		if len(h.PrivateAddress) == 0 {
+		if len(host.PrivateAddress) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath, "no private IP/address givevn"))
 		}
-		if len(h.SSHPrivateKeyFile) == 0 && len(h.SSHAgentSocket) == 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath, h.SSHPrivateKeyFile, "neither SSH private key nor agent socket given, don't know how to authenticate"))
-			allErrs = append(allErrs, field.Invalid(fldPath, h.SSHAgentSocket, "neither SSH private key nor agent socket given, don't know how to authenticate"))
+		if len(host.SSHPrivateKeyFile) == 0 && len(host.SSHAgentSocket) == 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath, host.SSHPrivateKeyFile, "neither SSH private key nor agent socket given, don't know how to authenticate"))
+			allErrs = append(allErrs, field.Invalid(fldPath, host.SSHAgentSocket, "neither SSH private key nor agent socket given, don't know how to authenticate"))
 		}
-		if len(h.SSHUsername) == 0 {
+		if len(host.SSHUsername) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath, "no SSH username given"))
 		}
-		if !h.OperatingSystem.IsValid() {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("operatingSystem"), h.OperatingSystem, "invalid operatingSystem provided"))
+		if !host.OperatingSystem.IsValid() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("operatingSystem"), host.OperatingSystem, "invalid operatingSystem provided"))
 		}
-		if h.Kubelet.MaxPods != nil && *h.Kubelet.MaxPods <= 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("kubelet").Child("maxPods"), h.Kubelet.MaxPods, "maxPods must be a positive number"))
+		if host.Kubelet.MaxPods != nil && *host.Kubelet.MaxPods <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("kubelet").Child("maxPods"), host.Kubelet.MaxPods, "maxPods must be a positive number"))
 		}
-		for labelKey, labelValue := range h.Labels {
+		for labelKey, labelValue := range host.Labels {
 			if strings.HasSuffix(labelKey, "-") && labelValue != "" {
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("labels"), labelValue, "label to remove cannot have value"))
 			}
 		}
-		if gte125Constraint.Check(v) {
-			for _, taint := range h.Taints {
-				if taint.Key == "node-role.kubernetes.io/master" {
-					allErrs = append(allErrs, field.Forbidden(fldPath.Child("taints"), fmt.Sprintf("%q taint is forbidden for clusters running Kubernetes 1.25+", "node-role.kubernetes.io/master")))
-				}
-			}
-		}
```

Comment:
I'm conflicted on this change. I would kind of prefer keeping this safeguard as long as manifests/addons tolerate taint for `node-role.kubernetes.io/master`. Once we remove this taint from manifests, removing this safeguard will be much safer. Until them, someone could try to apply this taint again for any reason (e.g. outdated KubeOneCluster manifest) and then run into problems once tolerations are removed. Also, kubeadm might be a little weird when it comes to this taint, so it further increases the risk of something going wrong.